### PR TITLE
Explicit :reasoning? opts for sparql/repo queries

### PR DESCRIPTION
**Pull request by andrewmcveigh**
Enable reasoning/inference to be turned on explicitly in an `opts` map for `grafter-2.rdf4j.repository/query` and `grafter-2.rdf4j.sparql/query`, defaulting to `off|false`.

Necessary to implement https://github.com/Swirrl/grafter.db/issues/1

### CODE REVIEW

**Comment by andrewmcveigh** [comment]
Because of https://github.com/eclipse/rdf4j/issues/784, we need to use a better repository/connection than `SPARQLProtocolSession` provides. Probably use something based on https://github.com/Swirrl/drafter/tree/master/src-java/drafter_sparql_repository/src/main/java/drafter/rdf

**Comment by andrewmcveigh** [comment]
Moved java `DrafterSparqlSession` etc., into `grafter.rdf`. Added simple HTTP server test to check parameters are being passed through.

File: src-java/grafter_sparql_repository/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Grafter SPARQL Repository
+
+This small piece of Java code enhances the SPARQL Repository to behave how we
+want it to when talking to Stardog.
```
Comment (RickMoynihan) [inline src-java/grafter_sparql_repository/README.md]: > when talking to Stardog

_when talking to SPARQL endpoints, such as Stardog_

**Comment by andrewmcveigh** [comment]
You mean here https://github.com/Swirrl/grafter/blob/605c0032d739f6f6cc30d1b9ea84752be06b5008/src/grafter_2/rdf4j/repository.clj#L141 ?

It should do. It's importing `grafter_2.rdf.SPARQLRepository` now, so might not be obvious.

**Comment by RickMoynihan** [comment]
Ahh yes I did mean there... was looking at the wrong branch - doh!  What you have looks great!  Soz.

**Pull request merged**